src/tree/updater_gpu_common.cuh

```diff
@@ -64,20 +64,13 @@ struct DeviceSplitCandidate {
   // split.
   bst_cat_t thresh{-1};
 
-  common::CatBitField split_cats;
```

Comment:
Trying to understand what this PR does.
Does the reference `split_cats` get invalidated as more categories are added? Why is it better to get the categories from the evaluator directly?

Reply:
Correct. It's invalidated when more nodes are added.

Reply:
So same reason as why you shouldn't carry around references to vector elements?  Got it.

Reply:
I think it was introduced during optimization and refactoring in 1.7. But should be fixed now. I will do some more tests myself. The error was only reproducible using a sanitizer.